src/backend/main.py

```diff
@@ -111,6 +112,20 @@ async def validation_exception_handler(request: Request, exc: Exception):
     )
 
 
+@app.exception_handler(DeploymentNotFoundError)
+async def deployment_not_found_handler(request: Request, exc: DeploymentNotFoundError):
+    ctx = get_context(request)
+    logger = ctx.get_logger()
+    logger.error(
+        event="Deployment not found",
+        deployment_id=exc.deployment_id,
+    )
+    return JSONResponse(
+        status_code=404,
```

Comment:
Probably should be error 400

Reply:
Yeah it's a tricky debate on whether 404 is appropriate for REST entities not being found or the URI itself being invalid. My opinion is that a 404 is fine if we provide additional context - which I think you are in this case